### PR TITLE
Impact para zoom plots

### PR DESCRIPTION
**Pull request by YonsiG**
This PR creates several zoom in plots for dxy and dz for muons

### CODE REVIEW

File: cpp/ScanChain_Zp.C [hunk 1]
```diff
@@ -292,11 +292,21 @@ int ScanChain(TChain *ch, double genEventSumw, TString year, TString process, in
     plot_names.push_back("mu1_phi");
     plot_names.push_back("mu2_phi");
     plot_names.push_back("mu1_dxy");
+    plot_names.push_back("mu1_dxyZoom");
```
Comment (VourMa) [inline cpp/ScanChain_Zp.C]: So as not to have a lot of output plots, I would propose to not add the zoom plots.

Instead, let's change the binning of the regular plots to what you have for the zoomed ones. The nominal plots are not useful as they are right now.

These holds for all the zoomed plots, not just dxyZoom.

File: cpp/ScanChain_Zp.C [hunk 2]
```diff
@@ -1511,6 +1600,9 @@ int ScanChain(TChain *ch, double genEventSumw, TString year, TString process, in
 
 	extra_plot_names.push_back("mu3_trkRelIso");
 	extra_variable.insert({"mu3_trkRelIso", Muon_tkRelIso.at(extra_muons[0])});
+
+  extra_plot_names.push_back("mu3_trkAbsIso");
```
Comment (VourMa) [inline cpp/ScanChain_Zp.C]: Could you please fix the indentation here?

File: cpp/configuration_Zp.h [hunk 3]
```diff
@@ -66,31 +66,81 @@ void histoDefinition(map<TString, int> &nbins, map<TString, float> &low, map<TSt
   high.insert({"mu1_dxy", 1.01});
   title.insert({"mu1_dxy", "d_{xy} (leading #mu)"});
 
+  nbins.insert({"mu1_dxyZoom", 200});
+  low.insert({"mu1_dxyZoom", 0.001});
+  high.insert({"mu1_dxyZoom", 0.21});
```
Comment (VourMa) [inline cpp/configuration_Zp.h]: I think that our upper boundary is 0.02, so there is still a lot of space wasted if we set the boundary at 0.21. Maybe move it to 0.021?

File: cpp/configuration_Zp.h [hunk 4]
```diff
@@ -66,31 +66,81 @@ void histoDefinition(map<TString, int> &nbins, map<TString, float> &low, map<TSt
...
+  title.insert({"mu1_dxyZoom", "d_{xy} (leading #mu)"});
+
   nbins.insert({"mu2_dxy", 100});
   low.insert({"mu2_dxy", 0.01});
   high.insert({"mu2_dxy", 1.01});
   title.insert({"mu2_dxy", "d_{xy} (subleading #mu)"});
 
+  nbins.insert({"mu2_dxyZoom", 200});
+  low.insert({"mu2_dxyZoom", 0.001});
+  high.insert({"mu2_dxyZoom", 0.21});
```
Comment (VourMa) [inline cpp/configuration_Zp.h]: Same as above.

File: cpp/configuration_Zp.h [hunk 5]
```diff
@@ -66,31 +66,81 @@ void histoDefinition(map<TString, int> &nbins, map<TString, float> &low, map<TSt
...
+  title.insert({"mu2_dxyZoom", "d_{xy} (leading #mu)"});
+
   nbins.insert({"mu1_dz", 100});
   low.insert({"mu1_dz", 0.01});
   high.insert({"mu1_dz", 1.01});
   title.insert({"mu1_dz", "d_{z} (leading #mu)"});
 
+  nbins.insert({"mu1_dzZoom", 500});
+  low.insert({"mu1_dzZoom", 0.001});
+  high.insert({"mu1_dzZoom", 0.501});
```
Comment (VourMa) [inline cpp/configuration_Zp.h]: The upper bound we apply is 0.1, so maybe move the upper histogram boundary to 0.101 as well?

File: cpp/configuration_Zp.h [hunk 6]
```diff
@@ -66,31 +66,81 @@ void histoDefinition(map<TString, int> &nbins, map<TString, float> &low, map<TSt
...
+  title.insert({"mu1_dzZoom", "d_{z} (leading #mu)"});
+
   nbins.insert({"mu2_dz", 100});
   low.insert({"mu2_dz", 0.01});
   high.insert({"mu2_dz", 1.01});
   title.insert({"mu2_dz", "d_{z} (subleading #mu)"});
 
+  nbins.insert({"mu2_dzZoom", 500});
+  low.insert({"mu2_dzZoom", 0.001});
+  high.insert({"mu2_dzZoom", 0.501});
```
Comment (VourMa) [inline cpp/configuration_Zp.h]: Same as above.

File: cpp/configuration_Zp.h [hunk 7]
```diff
@@ -66,31 +66,81 @@ void histoDefinition(map<TString, int> &nbins, map<TString, float> &low, map<TSt
...
+  title.insert({"mu2_dzZoom", "d_{z} (leading #mu)"});
+
   nbins.insert({"mu1_trkRelIso", 50});
   low.insert({"mu1_trkRelIso", 0.01});
   high.insert({"mu1_trkRelIso", 0.51});
   title.insert({"mu1_trkRelIso", "Track iso./p_{T} (leading #mu)"});
 
+  nbins.insert({"mu1_trkRelIsoZoom", 100});
+  low.insert({"mu1_trkRelIsoZoom", 0.001});
+  high.insert({"mu1_trkRelIsoZoom", 0.1});
```
Comment (VourMa) [inline cpp/configuration_Zp.h]: Move to 0.051 (we cut at 0.05)?

File: cpp/configuration_Zp.h [hunk 8]
```diff
@@ -66,31 +66,81 @@ void histoDefinition(map<TString, int> &nbins, map<TString, float> &low, map<TSt
...
+  title.insert({"mu1_trkRelIsoZoom", "Track iso./p_{T} (leading #mu)"});
+
+  nbins.insert({"mu1_trkAbsIso", 50});
+  low.insert({"mu1_trkAbsIso", 1.0});
+  high.insert({"mu1_trkAbsIso", 51});
+  title.insert({"mu1_trkAbsIso", "Track iso. (leading #mu)"});
+
+  nbins.insert({"mu1_trkAbsIsoZoom", 100});
+  low.insert({"mu1_trkAbsIsoZoom", 0.1});
+  high.insert({"mu1_trkAbsIsoZoom", 10.0});
```
Comment (VourMa) [inline cpp/configuration_Zp.h]: Move to 5.1 (we cut at 5)?

File: cpp/configuration_Zp.h [hunk 9]
```diff
@@ -66,31 +66,81 @@ void histoDefinition(map<TString, int> &nbins, map<TString, float> &low, map<TSt
...
+  title.insert({"mu1_trkAbsIsoZoom", "Track iso. (leading #mu)"});
+
   nbins.insert({"mu2_trkRelIso", 50});
   low.insert({"mu2_trkRelIso", 0.01});
   high.insert({"mu2_trkRelIso", 0.51});
   title.insert({"mu2_trkRelIso", "Track iso./p_{T} (subleading #mu)"});
 
+  nbins.insert({"mu2_trkRelIsoZoom", 100});
+  low.insert({"mu2_trkRelIsoZoom", 0.001});
+  high.insert({"mu2_trkRelIsoZoom", 0.1});
```
Comment (VourMa) [inline cpp/configuration_Zp.h]: Same as for mu1.

File: cpp/configuration_Zp.h [hunk 10]
```diff
@@ -66,31 +66,81 @@ void histoDefinition(map<TString, int> &nbins, map<TString, float> &low, map<TSt
...
+  title.insert({"mu2_trkRelIsoZoom", "Track iso./p_{T} (leading #mu)"});
+
+  nbins.insert({"mu2_trkAbsIso", 50});
+  low.insert({"mu2_trkAbsIso", 1.0});
+  high.insert({"mu2_trkAbsIso", 51});
+  title.insert({"mu2_trkAbsIso", "Track iso. (subleading #mu)"});
+
+  nbins.insert({"mu2_trkAbsIsoZoom", 100});
+  low.insert({"mu2_trkAbsIsoZoom", 0.1});
+  high.insert({"mu2_trkAbsIsoZoom", 10.0});
```
Comment (VourMa) [inline cpp/configuration_Zp.h]: Same as for mu1.

File: cpp/doAll_Zp.C [hunk 11]
```diff
@@ -29,6 +29,12 @@
   map<TString,TString> sample_names = { };
   map<TString,map<TString,vector<TString>>> sample_prod = { };
 
+  bool run_data = 0;
```
Comment (VourMa) [inline cpp/doAll_Zp.C]: Can you make the default to be to run everything (to keep the same behavior as now)?

File: cpp/doAll_Zp.C [hunk 12]
```diff
@@ -29,6 +29,12 @@
...
+  bool run_MCbkg = 0;
+  bool run_signal = 1;
+//  bool pick_signal_mass = 0;
```
Comment (VourMa) [inline cpp/doAll_Zp.C]: Maybe remove the comment, since it is not used anywhere?

File: cpp/doAll_Zp.C [hunk 13]
```diff
@@ -156,24 +164,31 @@
...
 
   // Signals
-  vector<TString> sigModel = { "Y3", "DY3", "DYp3", "B3mL2" };
-  vector<TString> sigMass = { /*"100",*/ "200", "400", "700", "1000", "1500", "2000" };
-  for ( unsigned int imodel=0; imodel<sigModel.size(); imodel++ )
-  {
-    for ( unsigned int imass=0; imass<sigMass.size(); imass++ )
+  if(run_signal){
+    vector<TString> sigModel = { "Y3", "DY3", "DYp3", "B3mL2" };
+//    if (pick_signal_mass) {
```
Comment (VourMa) [inline cpp/doAll_Zp.C]: Same comment about comments.

File: cpp/doAll_Zp.C [hunk 14]
```diff
@@ -29,6 +29,12 @@
   map<TString,TString> sample_names = { };
   map<TString,map<TString,vector<TString>>> sample_prod = { };
 
+  bool run_data = 0;
+  bool run_MCbkg = 0;
+  bool run_signal = 1;
+//  bool pick_signal_mass = 0;
+
+  if(run_data){
```
Comment (VourMa) [inline cpp/doAll_Zp.C:36]: Can you fix the indentation for the following lines, like you did for the signal?

File: cpp/doAll_Zp.C [hunk 15]
```diff
@@ -60,7 +66,9 @@
 						 "Run2016G-UL2016_MiniAODv2_NanoAODv9-v1",
 						 "Run2016H-UL2016_MiniAODv2_NanoAODv9-v1",
 						} } } });
+  }
 
+  if(run_MCbkg){
```
Comment (VourMa) [inline cpp/doAll_Zp.C:70]: Same comment as above.

File: cpp/doAll_Zp.C [hunk 16]
```diff
@@ -156,24 +164,31 @@
...
+  if(run_signal){
+    vector<TString> sigModel = { "Y3", "DY3", "DYp3", "B3mL2" };
+//    if (pick_signal_mass) {
+//      vector<TString> sigMass = { "200", "700", "1500" };
+//    }
+//    else {
+      vector<TString> sigMass = { /*"100",*/ "200", "400", "700", "1000", "1500", "2000" };
+//    }
+    for ( unsigned int imodel=0; imodel<sigModel.size(); imodel++ )
     {
```
Comment (VourMa) [inline cpp/doAll_Zp.C:173]: Indentation needs to be fixed here?

File: cpp/doAll_Zp.C [hunk 17]
```diff
@@ -156,24 +164,31 @@
...
-      samples.push_back(sigModel[imodel]+"_M"+sigMass[imass]);
-      sample_names.insert({sigModel[imodel]+"_M"+sigMass[imass],"ZPrimeToMuMuSB_M"+sigMass[imass]+"_bestfit_TuneCP5_13TeV_Allanach_"+sigModel[imodel]+"_5f_madgraph_pythia8_NoPSWgts"});
-      sample_prod.insert({sigModel[imodel]+"_M"+sigMass[imass], { { "2018",       { "RunIISummer20UL18MiniAODv2-106X_upgrade2018_realistic_v16_L1v1-v2_private" } },
+      for ( unsigned int imass=0; imass<sigMass.size(); imass++ )
+      {
+        samples.push_back(sigModel[imodel]+"_M"+sigMass[imass]);
+        sample_names.insert({sigModel[imodel]+"_M"+sigMass[imass],"ZPrimeToMuMuSB_M"+sigMass[imass]+"_bestfit_TuneCP5_13TeV_Allanach_"+sigModel[imodel]+"_5f_madgraph_pythia8_NoPSWgts"});
+        sample_prod.insert({sigModel[imodel]+"_M"+sigMass[imass], { { "2018",       { "RunIISummer20UL18MiniAODv2-106X_upgrade2018_realistic_v16_L1v1-v2_private" } },
                                                                   { "2017",       { "RunIISummer20UL17MiniAODv2-106X_mc2017_realistic_v9-v2_private" } },
```
Comment (VourMa) [inline cpp/doAll_Zp.C:179]: Indentation needs to be fixed here (to match the format of other samples)?

File: cpp/ScanChain_Zp.C (hunk 1 above)
Comment (YonsiG) [reply cpp/ScanChain_Zp.C]: fixed

File: cpp/configuration_Zp.h (hunk 3 above)
Comment (YonsiG) [reply cpp/configuration_Zp.h]: fixed

File: cpp/configuration_Zp.h (hunk 4 above)
Comment (YonsiG) [reply cpp/configuration_Zp.h]: fixed

File: cpp/configuration_Zp.h (hunk 5 above)
Comment (YonsiG) [reply cpp/configuration_Zp.h]: fixed

File: cpp/configuration_Zp.h (hunk 6 above)
Comment (YonsiG) [reply cpp/configuration_Zp.h]: fixed

File: cpp/configuration_Zp.h (hunk 7 above)
Comment (YonsiG) [reply cpp/configuration_Zp.h]: fixed

File: cpp/configuration_Zp.h (hunk 8 above)
Comment (YonsiG) [reply cpp/configuration_Zp.h]: fixed

File: cpp/configuration_Zp.h (hunk 9 above)
Comment (YonsiG) [reply cpp/configuration_Zp.h]: fixed

File: cpp/configuration_Zp.h (hunk 10 above)
Comment (YonsiG) [reply cpp/configuration_Zp.h]: fixed

File: cpp/doAll_Zp.C (hunk 11 above)
Comment (YonsiG) [reply cpp/doAll_Zp.C]: fixed

File: cpp/doAll_Zp.C (hunk 12 above)
Comment (YonsiG) [reply cpp/doAll_Zp.C]: fixed

File: cpp/doAll_Zp.C (hunk 13 above)
Comment (YonsiG) [reply cpp/doAll_Zp.C]: fixed

File: cpp/ScanChain_Zp.C (hunk 2 above)
Comment (YonsiG) [reply cpp/ScanChain_Zp.C]: fixed

File: cpp/doAll_Zp.C (hunk 14 above)
Comment (YonsiG) [reply cpp/doAll_Zp.C:36]: fixed

File: cpp/doAll_Zp.C (hunk 15 above)
Comment (YonsiG) [reply cpp/doAll_Zp.C:70]: fixed

File: cpp/doAll_Zp.C (hunk 16 above)
Comment (YonsiG) [reply cpp/doAll_Zp.C:173]: fixed

**Comment by VourMa** [comment]
The PR looks great now, thanks! Merging...

**Pull request merged**